### DISCUSSION TITLE
duration of event

### DISCUSSION THREAD

**Discussion by xabierr** [georgebv/pyextremes#63] [Q&A]
Hi there,

Is there any way to define the duration of an event over a threshold? For example, 24 hours consecutively above a threshold value?

**Comment by georgebv** [comment]
Can you please explain in more detail how this would work? Are you looking for periods of at least 24 hours where all observations exceed the threshold? If so, could this be a pre-processing step? To me this sounds like removal of outliers or denoising.

I know this doesn't quite fit what you ask, but pyextremes has declustering parameter `r` when extracting POT extremes https://georgebv.github.io/pyextremes/user-guide/4-peaks-over-threshold/#declustering. This is based on the IID (independent and identically distributed) requirement of extremes when fitting a model - extreme values within a cluster are likely dependent.

**Reply by xabierr** [reply]
I need to count how many storms in a time series. A storm is defined as an event lasting at least 24 hours over a threshold of 3.

**Reply by georgebv** [reply]
How would this work in this scenario: you have 23 hours of water level above 3, then it dips below 3 for 1 hour, then it's above 3 for another 10 hours. Would you discard this event?

**Comment by xabierr** [comment]
Yes

Sent on the go with Vodafone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
